src/Streamly/Internal/Unicode/Utf8/Eliminate.hs

```diff
@@ -0,0 +1,414 @@
+-- |
+-- Module      : Streamly.Internal.Unicode.Utf8.Eliminate
+-- Copyright   : (c) 2021 Composewell Technologies
+-- License     : BSD-3-Clause
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+--
+module Streamly.Internal.Unicode.Utf8.Eliminate
+    (
+    -- * Folds
+      fold
+    , foldl
+    , foldl'
+    , foldl1
+    , foldl1'
+    , foldr
+    , foldr1
+
+    -- ** Special folds
+    , any
+    , all
+    , maximum
+    , minimum
+
+    -- * Predicates
+
+    -- XXX Comment our further stuff to help with the GHCJS compilation
+    -- , isPrefixOf
+    -- , isSuffixOf
+    -- XXX isInfixOf takes too much memory to compile
+    -- , isInfixOf
+
+    -- ** View patterns
+
+    -- XXX Comment our further stuff to help with the GHCJS compilation
+    -- , stripPrefix
+    -- , stripSuffix
+    , commonPrefixes
+
+    -- * Searching
+    , find
+    , elem
+
+    -- * Indexing
+    -- $index
+    , index
+    , findIndex
+    , countChar
+    , count
+    )
+where
+
+#include "inline.hs"
+
+--------------------------------------------------------------------------------
+-- Imports
+--------------------------------------------------------------------------------
+
+import Control.Monad.IO.Class (MonadIO(..))
+import Streamly.Internal.Data.Fold (Fold)
+import Streamly.Internal.Data.Stream.IsStream.Lift (hoist)
+import System.IO.Unsafe (unsafePerformIO)
+
+import qualified Streamly.Internal.Data.Array.Foreign.Type as Array
+import qualified Streamly.Internal.Data.Stream.IsStream as Stream
+
+import Streamly.Internal.Unicode.Utf8.Type
+
+import Prelude hiding
+    ( all
+    , any
+    , break
+    , concat
+    , concatMap
+    , drop
+    , dropWhile
+    , elem
+    , filter
+    , foldl
+    , foldl1
+    , foldr
+    , foldr1
+    , head
+    , init
+    , last
+    , length
+    , lines
+    , map
+    , maximum
+    , minimum
+    , null
+    , read
+    , replicate
+    , reverse
+    , scanl
+    , scanl1
+    , scanr
+    , scanr1
+    , span
+    , splitAt
+    , tail
+    , take
+    , takeWhile
+    , unlines
+    , unwords
+    , words
+    , zip
+    , zipWith
+    )
+
+-- $setup
+-- >>> :set -XOverloadedStrings
+-- >>> import qualified Streamly.Internal.Unicode.Utf8 as Utf8
+
+--------------------------------------------------------------------------------
+-- Reducing Streams (folds)
+--------------------------------------------------------------------------------
+
+{-# INLINE fold #-}
+fold :: MonadIO m => Fold m Char b -> Utf8 -> m b
+fold f = Stream.fold f . hoist liftIO . toStream
+
+-- | 'foldl', applied to a binary operator, a starting value
+-- (typically the left-identity of the operator), and a 'Utf8',
+--
+-- /Time complexity:/ O(n)
+-- reduces the 'Utf8' using the binary operator, from left to right.
+--
+{-# INLINE foldl #-}
+foldl :: (a -> Char -> a) -> a -> Utf8 -> a
+foldl = undefined
```

Comment:
Can you add the `/Unimplemented/` in the docs for undefined functions so that it appears in the docs?